server/engagement/models.py

```diff
@@ -0,0 +1,25 @@
+from django.db import models
+from members.models import User
+
+class AttendanceSession(models.Model):
+    id = models.AutoField(primary_key=True)
+    key = models.CharField(max_length=10) # session keys don't have to be unique
```

Comment:
they will need to be unique as far as unexpired sessions, unless we want the expected behavior to be "create a record for every session with corresponding key". Fine with either, or with just verifying against sessions based on expiration, in which case we might want a sort index on this field

Reply:
```py
class ModelType(models.Model):
  class Meta:
    indexes = [models.Index(fields=['-expires'])]
```